apiserver/authentication/user.go

```diff
@@ -287,8 +271,27 @@ func (m *ExternalMacaroonAuthenticator) Authenticate(entityFinder EntityFinder,
 	return entity, nil
 }
 
-func addMacaroonTimeBeforeCaveat(svc BakeryService, m *macaroon.Macaroon, t time.Time) error {
-	return svc.AddCaveat(m, checkers.TimeBeforeCaveat(t))
+// IdentityFromContext implements IdentityClient.IdentityFromContext.
+func (m *ExternalMacaroonAuthenticator) IdentityFromContext(ctx context.Context) (identchecker.Identity, []checkersv2.Caveat, error) {
+	expiryTime := time.Now().Add(externalLoginExpiryTime)
```

Comment:
Why use time.Now() and not a clock?

Reply:
This code was just moved from one place to another to fit the newer bakery methods. I'll see if a clock can be used.